src/main/java/org/checkerframework/specimin/TargetMethodFinderVisitor.java

```diff
@@ -200,10 +221,21 @@ public Visitable visit(ExplicitConstructorInvocationStmt expr, Void p) {
 
   @Override
   public Visitable visit(FieldAccessExpr expr, Void p) {
+    if (insideTargetMethod) {
+      usedMethods.add(classFQName + "#" + expr.getName().asString());
+    }
     Expression caller = expr.getScope();
     if (caller instanceof SuperExpr) {
       usedClass.add(caller.calculateResolvedType().describe());
     }
     return super.visit(expr, p);
   }
+
+  @Override
+  public Visitable visit(NameExpr expr, Void p) {
+    if (!declaredNames.contains(expr.getNameAsString())) {
+      usedClass.add(parentClass);
```

Comment:
This concerns me, because I don't think we have a guarantee that all variable declarators are visited before all names. In particular, the following is legal Java code:
```
public class OutOfOrder {
    public static void use() {
	System.out.println(a);
    }

    static String a = "hello world";
}
```

In this code, `a` is used before it is declared, so I think Specimin might be tricked by this case. Can you add a test case like this to make sure that your approach works, even on code with a non-standard order of declarations?

Reply:
Thank for pointing this out. I did not realize that a variable can be used before it is declared. I will try to think of another way to get around this.